mnefun/_mnefun.py

```diff
@@ -1419,7 +1419,21 @@ def save_epochs(p, subjects, in_names, in_numbers, analyses, out_names,
             evokeds = list()
             for name in names:
                 if p.average_weighting == 'chpi':
-                    weights = _chpi_snr_epochs(e[name])
+                    # read in raw unfiltered files for cHPI weighting evokeds
+                    raw_names = get_raw_fnames(p, subj, 'raw', False, True,
+                                               run_indices[si])
+                    raw = [Raw(fname, preload=False, allow_maxshield=True)
+                           for fname in raw_names]
+                    raw_unfil = concatenate_raws(raw)
+                    eps_unfil = Epochs(raw_unfil, events=e[name].events,
+                                       event_id=None, tmin=p.tmin,
+                                       tmax=p.tmax, baseline=_get_baseline(p),
+                                       reject=None, flat=None,
+                                       proj=None, preload=True,
+                                       decim=decim[si], picks=e.picks,
+                                       on_missing='warning')
+                    # Use unfiltered epochs for weights
```

Comment:
@jjnurminen I am pretty sure that `eps_unfil` is what we want in terms of unfiltered raw data epoched as per user defined conditions that is subsequently passed to `_chpi_snr_epochs`.
@Eric89GXL Please vet this one in terms of sampling/filtering, I am pretty confident it's working as expected but I would appreciate a second opinion.